packages/playground/src/components/app-root/app-root.tsx

```diff
@@ -417,12 +419,33 @@ export class AppRoot {
   }
 
   waitForMultisig() {
-    const { user } = this.accountState;
+    const {
+      user: { transactionHash }
+    } = this.accountState;
+
     const provider = this.walletState.provider as Web3Provider;
 
-    provider.once(user.transactionHash, async () => {
-      await this.fetchMultisig();
-    });
+    let onMultisigMinedHasBeenCalled = false;
+    const onMultisigMined = async () => {
+      if (!onMultisigMinedHasBeenCalled) {
+        await this.fetchMultisig();
+        onMultisigMinedHasBeenCalled = true;
+      }
+    };
+
+    provider.once(transactionHash, onMultisigMined);
+
+    setTimeout(() => {
+      if (!onMultisigMinedHasBeenCalled) {
+        console.log("Tx event not emitted within 24s, polling every 5s now");
+        const poll = setInterval(async () => {
```

Comment:
i generally recommend avoiding setInterval in favor of setTimeout's that schedule another setTimeout. this tends to be better for testing.